provider/common/errors_test.go

```diff
@@ -32,3 +32,18 @@ github.com/juju/juju/provider/common/errors_test.go:.*: foo
 github.com/juju/juju/provider/common/errors_test.go:.*: bar
 github.com/juju/juju/provider/common/errors_test.go:.*: bar: foo`[1:])
 }
+
+func (s *ErrorsSuite) TestNewInvalidCredential(c *gc.C) {
+	err1 := errors.New("foo")
+	err2 := errors.Annotate(err1, "bar")
+	err := common.CredentialNotValid(err2)
+
+	c.Assert(err, jc.Satisfies, common.IsCredentialNotValid)
```

Comment:
should we have the inverse assertion as well? Something that clearly *doesn't* satisfy IsCredentalNotValid ? (weeds out the "return true" implementation)